daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionCoderSpec.scala

```diff
@@ -446,6 +446,95 @@ class TransactionCoderSpec
 
   }
 
+  "decodeNodeVersion" should {
+
+    val postV10versions = TransactionVersion.All.filter(_ > V10)
+
+    "succeed as expected when the node is encoded with an version older than the transaction version" in {
+
+      forAll(danglingRefGenNode, versionInIncreasingOrder(postV10versions), minSuccessful(5)) {
+        case ((nodeId, node), (v1, v2)) =>
+          val normalizedNode = normalizeNode(node.updateVersion(v1))
+
+          val Right(encoded) = TransactionCoder
+            .encodeNode(
+              TransactionCoder.NidEncoder,
+              ValueCoder.CidEncoder,
+              v1,
+              nodeId,
+              normalizedNode,
+            )
+
+          TransactionCoder.decodeNodeVersion(v2, encoded) shouldBe Right(v1)
+      }
+    }
+
+    "fail when the node is encoded with a version newer than the transaction version" in {
+
+      forAll(
+        danglingRefGenNode,
+        versionInStrictIncreasingOrder(postV10versions),
+        minSuccessful(5),
+      ) { case ((nodeId, node), (v1, v2)) =>
+        val normalizedNode = normalizeNode(node.updateVersion(v2))
+
+        val Right(encoded) = TransactionCoder
+          .encodeNode(
+            TransactionCoder.NidEncoder,
+            ValueCoder.CidEncoder,
+            v2,
+            nodeId,
+            normalizedNode,
+          )
+
+        TransactionCoder.decodeNodeVersion(v1, encoded) shouldBe a[Left[_, _]]
+      }
+    }
+
+    "return V10 when the node is not version" in {
+
+      forAll(danglingRefGenNode, minSuccessful(5)) { case (nodeId, node) =>
+        val normalizedNode = normalizeNode(node.updateVersion(V10))
+
+        val Right(encoded) = TransactionCoder
+          .encodeNode(
+            TransactionCoder.NidEncoder,
+            ValueCoder.CidEncoder,
+            V10,
+            nodeId,
+            normalizedNode,
+          )
+
+        assert(encoded.getVersion.isEmpty)
+
+        TransactionCoder.decodeNodeVersion(V10, encoded) shouldBe Right(V10)
+      }
+    }
+
+    "ignore node version field when the transaction in transaction version < 11" in {
```

Comment:
```suggestion
    "ignore node version field when transaction version = 10" in {
```
(because we don't check any other version)